packages/components/Shimmer/src/Shimmer.tsx

```diff
@@ -0,0 +1,98 @@
+/** @jsx withSlots */
+import { compose, UseSlots, buildProps, mergeProps, withSlots } from '@fluentui-react-native/framework';
+import { ViewProps } from 'react-native';
+import { IViewProps } from '@fluentui-react-native/adapters';
+import * as React from 'react';
+import { ensureNativeComponent } from '@fluentui-react-native/component-cache';
+
+const shimmerName = 'Shimmer';
+
+const NativeShimmerView = ensureNativeComponent('MSFShimmerView');
+
+/**
+ * ViewProps props, based off of the standard react-native ViewProps with some new extensions
+ */
+export type NativeShimmerViewProps<TBase = IViewProps> = TBase & {
+  /**
+   * The alpha value of the center of the gradient in the animation
+   */
+  shimmerAlpha?: number;
+
+  /**
+   * The width of the gradient in the animation
+   */
+  shimmerWidth?: number;
+
+  /**
+   * Angle of the direction of the gradient, in radian. 0 means horizontal, Pi/2 means vertical.
+   */
+  shimmerAngle?: number;
+
+  /**
+   * Speed of the animation, in point/seconds.
+   */
+  shimmerSpeed?: number;
+
+  /**
+   * Delay between the end of a shimmering animation and the beginning of the next one, in seconds
+   */
+  shimmerDelay?: number;
+
+  /**
+   * Color to tint the shimmer boxes. Defaults to the Fabric default color.
+   */
+  viewTintColor?: string;
+
+  /**
+   * Corner radius on each view.
+   */
+  cornerRadius?: number;
+};
+
+/**
+ * Shimmer Token is a flat list of all the properties we can set on the native view
+ */
+export type ShimmerToken = NativeShimmerViewProps;
+
+interface ShimmerType {
+  props: ViewProps;
+  slotProps: { root: NativeShimmerViewProps };
+  tokens: ShimmerToken;
+}
+
+export const Shimmer = compose<ShimmerType>({
+  displayName: shimmerName,
+  tokens: [
+    {
+      shimmerAlpha: 0.4,
+      shimmerWidth: 180,
+      shimmerAngle: -(Math.PI / 45.0),
+      shimmerSpeed: 350,
+      shimmerDelay: 0.4,
+      viewTintColor: 'rgb(241,241,241)',
+      cornerRadius: 4,
+    },
+    shimmerName,
+  ],
```

Comment:
You should add either `tokenProps: true` which will just copy all props into tokens, or `tokenProps: ['shimmerAlpha', 'shimmerWidth', // and so on. The second approach will copy just those properties into tokens, at which point you could likely change your buildProps to just spread the tokens.  Easiest way would be to do something like the following:
```ts
const tokenProps: (keyof ShimmerTokens)[] = [/* list all the tokens here */];

// in the body of the component then add
tokenProps,

// then buildProps becomes:
  root: buildProps(tokens => ({...tokens}), tokenProps)
```

Reply:
I'm gonna keep this example around somewhere when I inevitably have to write a guide for converting more FluentUI-Apple controls..